File: Scripts/javascript/shardeum_testing/eventPolling/ShardeumCycleTransactionPageReadExample.js
const axios = require('axios');

let baseUrl = "https://explorer-sphinx.shardeum.org/api/transaction?startCycle=49330&endCycle=49330&address=0x6bd9e67bf927da1935b6eaea9bf22500c4e1f53a"

getTransactionsToAddressCycleRange(baseUrl)

async function readJSONLoop(totalTransactions) {

	let total = totalTransactions;
	let pageIndex = 1

	while ( total > 0 ) {

		let filterUrl = baseUrl + "&page=" + pageIndex
		console.log(filterUrl)

		let responseRawJSON = await axios.get(filterUrl);
		responseRawJSON = responseRawJSON.data;
		console.log(responseRawJSON);

	  	total -= 10;
		pageIndex++;
	}

}

async function getTransactionsToAddressCycleRange(baseUrl) {

	let responseRawJSON = await axios.get(baseUrl);
	let responseDataJSON = responseRawJSON.data;
	let totalTransactions = responseDataJSON.totalTransactions
	console.log(totalTransactions);

	readJSONLoop(totalTransactions)
    
}

File: Scripts/zOtherLanguages/python/shardeum/ShardeumCycleTransactionPageReadExample.py
from urllib.request import urlopen
import json

transactionsInCycleRangeUrlString = "https://explorer.liberty20.shardeum.org/api/transaction?startCycle=0&endCycle=1000&address=0x0000000000000000000000000000000000000000"
transactionsInCycleRangeUrlOpened = urlopen(transactionsInCycleRangeUrlString)
transactionsInCycleRangeUrlJSON = json.loads(transactionsInCycleRangeUrlOpened.read())
totalTransactions = transactionsInCycleRangeUrlJSON["totalTransactions"]
print(totalTransactions)
pageIndex = 1

while totalTransactions > 0:
    print(pageIndex)
    print(totalTransactions)
    pageIndexIncrementUrlString = transactionsInCycleRangeUrlString + "&page=" + str(pageIndex)
    pageIndexIncrementUrlOpened = urlopen(pageIndexIncrementUrlString)
    rawTransactionDataPage = json.loads(pageIndexIncrementUrlOpened.read())
    print(rawTransactionDataPage)
    totalTransactions -= 10
    pageIndex += 1
